src/phases/mystery-encounter-phases.ts

```diff
@@ -0,0 +1,606 @@
+      }
+    }
+
+    const availablePartyMembers = scene.getParty().filter(p => !p.isFainted());
+
+    if (!availablePartyMembers[0].isOnField()) {
+      scene.pushPhase(new SummonPhase(scene, 0));
+    }
+
+    if (scene.currentBattle.double) {
+      if (availablePartyMembers.length > 1) {
+        scene.pushPhase(new ToggleDoublePositionPhase(scene, true));
+        if (!availablePartyMembers[1].isOnField()) {
+          scene.pushPhase(new SummonPhase(scene, 1));
+        }
+      }
+    } else {
+      if (availablePartyMembers.length > 1 && availablePartyMembers[1].isOnField()) {
+        scene.pushPhase(new ReturnPhase(scene, 1));
+      }
+      scene.pushPhase(new ToggleDoublePositionPhase(scene, false));
+    }
+
+    if (encounterMode !== MysteryEncounterMode.TRAINER_BATTLE && !this.disableSwitch) {
+      const minPartySize = scene.currentBattle.double ? 2 : 1;
+      if (availablePartyMembers.length > minPartySize) {
+        scene.pushPhase(new CheckSwitchPhase(scene, 0, scene.currentBattle.double));
+        if (scene.currentBattle.double) {
+          scene.pushPhase(new CheckSwitchPhase(scene, 1, scene.currentBattle.double));
+        }
+      }
+    }
+
+    this.end();
+  }
+
+  /**
+   * Ease in enemy trainer
+   * @private
+   */
+  private showEnemyTrainer(): void {
+    // Show enemy trainer
+    const trainer = this.scene.currentBattle.trainer;
+    if (!trainer) {
+      return;
+    }
+    trainer.alpha = 0;
+    trainer.x += 16;
+    trainer.y -= 16;
+    trainer.setVisible(true);
+    this.scene.tweens.add({
+      targets: trainer,
+      x: "-=16",
+      y: "+=16",
+      alpha: 1,
+      ease: "Sine.easeInOut",
+      duration: 750,
+      onComplete: () => {
+        trainer.untint(100, "Sine.easeOut");
+        trainer.playAnim();
+      }
+    });
+  }
+
+  private hideEnemyTrainer(): void {
+    this.scene.tweens.add({
+      targets: this.scene.currentBattle.trainer,
+      x: "+=16",
+      y: "-=16",
+      alpha: 0,
+      ease: "Sine.easeInOut",
+      duration: 750
+    });
+  }
+}
+
+/**
+ * Will handle (in order):
+ * - doContinueEncounter() callback for continuous encounters with back-to-back battles (this should push/shift its own phases as needed)
+ *
+ * OR
+ *
+ * - Any encounter reward logic that is set within MysteryEncounter doEncounterExp
+ * - Any encounter reward logic that is set within MysteryEncounter doEncounterRewards
+ * - Otherwise, can add a no-reward-item shop with only Potions, etc. if addHealPhase is true
+ * - Queuing of the PostMysteryEncounterPhase
+ */
+export class MysteryEncounterRewardsPhase extends Phase {
+  addHealPhase: boolean;
+
+  constructor(scene: BattleScene, addHealPhase: boolean = false) {
+    super(scene);
+    this.addHealPhase = addHealPhase;
+  }
+
+  /**
+   * Runs {@link MysteryEncounter.doContinueEncounter} and ends phase, OR {@link MysteryEncounter.onRewards} then continues encounter
```

Comment:
```suggestion
   * Runs `MysteryEncounter.doContinueEncounter` and ends phase, OR `MysteryEncounter.onRewards` then continues encounter
```
Unfortunately that doesn't work for things that aren't imported; and while it's "possible" to do comment-only imports, they seem to be buggy (or at least the extension's handling of them is buggy).